swift/Canvas.swift

```diff
@@ -0,0 +1,53 @@
+import CoreGraphics
+
+import thorvg
+
+/// A Swift wrapper for ThorVG's Canvas, facilitating drawing operations.
+class Canvas {
+    /// Pointer to the underlying ThorVG canvas object.
+    let pointer: OpaquePointer
+
+    /// The size of the canvas.
+    let size: CGSize
+
+    /// A flag indicating whether the canvas is empty.
+    var isEmpty: Bool = true
+
+    /// Initializes a canvas with a specific size, buffer, and stride for drawing.
+    init(size: CGSize, buffer: Buffer, stride: Int) {
+        self.pointer = tvg_swcanvas_create()
+        self.size = size
+
+        tvg_swcanvas_set_target(pointer, buffer, UInt32(stride), UInt32(size.width), UInt32(size.height), TVG_COLORSPACE_ARGB8888)
+    }
+
+    /// Pushes a picture onto the the canvas.
+    func push(picture: Picture) throws {
+        tvg_canvas_push(pointer, picture.pointer)
+        self.isEmpty = false
+    }
+
+    /// Updates the properties of a picture that has already been pushed onto the canvas.
+    func update(picture: Picture) {
+        tvg_canvas_update_paint(pointer, picture.pointer)
+    }
+
+    /// Clears the canvas without deallocating resources.
+    func clear() {
+        tvg_canvas_clear(pointer, false)
+    }
+
+    /// Draws the contents of the canvas into the buffer.
+    func draw() throws {
+        guard tvg_canvas_draw(pointer) == TVG_RESULT_SUCCESS else {
+            throw ThorVGError.failedToDrawFrame
+        }
+
+        tvg_canvas_sync(pointer)
+    }
+
+    /// Destroys the canvas and deallocates the associated resources.
+    func destroy() {
+        tvg_canvas_destroy(pointer)
+    }
```

Comment:
Great find. I was getting some issues when calling this `tvg_canvas_destroy` in the `Canvas` `deinit` but now I realise that was because I was also calling it from the `LottieRenderer`. 

I'll just call it once from the `deinit` of the `Canvas` 👍